dbms/src/AggregateFunctions/AggregateFunctionAvg.h

```diff
@@ -103,9 +89,33 @@ class AggregateFunctionAvg final : public IAggregateFunctionDataHelper<Data, Agg
 
     const char * getHeaderFilePath() const override { return __FILE__; }
 
-private:
+protected:
     UInt32 scale;
 };
 
+template <typename T, typename Data>
+class AggregateFunctionAvg final : public AggregateFunctionAvgBase<Data, T, AggregateFunctionAvg<T, Data>>
+{
+public:
+    AggregateFunctionAvg(const DataTypes & argument_types_)
```

Comment:
You can probably inherit constructor from `AggregateFunctionAvgBase`. It's not very important and not necessary to fix.